### PR TITLE
Add user activation API

### PR DESCRIPTION
**Pull request by foolip**
This is a Baseline 2023 feature.

### CODE REVIEW

File: feature-group-definitions/user-activation.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+name: User activation
+description: The `navigator.userActivation` API reveals whether the user has interacted with the page through an "activation" gesture such as a click, tap, or keyboard shortcut. This can be useful when working with APIs that require user activation, such as the fullscreen API.
```
Comment (foolip) [inline feature-group-definitions/user-activation.yml:2]: @mustaqahmed can you help me review this description? I've tried to explain the use case but am not sure I got it right. Is the main purpose of the API to predict when a call will fail, to provide feature detection for the new user activation model, or something else?

**Review by ddbeck — APPROVED** [review]

File: feature-group-definitions/user-activation.yml (hunk 1 above)
Comment (mustaqahmed) [reply feature-group-definitions/user-activation.yml:2]: Thanks for bringing this to my attention.  I made some suggestions above.

File: feature-group-definitions/user-activation.dist.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Generated from: user-activation.yml
+# Do not edit this file by hand. Edit the source file instead!
+
+name: User activation
+description: The `navigator.userActivation` API reveals whether the user has interacted with the page through an "activation" gesture such as a click, tap, or keyboard shortcut. This can be useful when working with APIs that require user activation, such as the fullscreen API.
```
[the remark this answers is not in the document]
Comment (foolip) [reply feature-group-definitions/user-activation.dist.yml:5]: Thanks, I wasn't sure what to call pressing the spacebar or enter key as opposed to any key press.

**Comment by foolip** [comment]
Thank you @mustaqahmed! I've pushed changes following your suggestions.

File: feature-group-definitions/user-activation.dist.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (foolip) [reply feature-group-definitions/user-activation.dist.yml]: I took that, just changing it from talking about developers to instead speak to developers.

**Review by mustaqahmed — APPROVED** [review]

**Pull request merged**